lib/valueflow.cpp

```diff
@@ -1345,7 +1345,19 @@ static void valueFlowRightShift(TokenList *tokenList)
             continue;
         if (lhsmax < 0)
             continue;
-        if ((1 << rhsvalue) <= lhsmax)
+        int lhsbits;
+        if ((tok->astOperand1()->valueType()->type == ValueType::Type::CHAR) ||
+            (tok->astOperand1()->valueType()->type == ValueType::Type::SHORT) ||
+            (tok->astOperand1()->valueType()->type == ValueType::Type::BOOL) ||
+            (tok->astOperand1()->valueType()->type == ValueType::Type::INT))
+            lhsbits = settings->int_bit;
+        else if (tok->astOperand1()->valueType()->type == ValueType::Type::LONG)
+            lhsbits = settings->long_bit;
+        else if (tok->astOperand1()->valueType()->type == ValueType::Type::LONGLONG)
+            lhsbits = settings->long_long_bit;
```

Comment:
This is (more or less) copy-pasted from lib/checktype.cpp. It should probably be extracted to a function, but I didn't really find a good place to put it, where should such a function be located?